net/etesync-server/files/init.uwsgi

```diff
@@ -0,0 +1,117 @@
+#!/bin/ash /etc/rc.common
+
+START=80
+
+USE_PROCD=1
+
+INI="; this file is autmatically re-created from /etc/config/etesync_server\n"
+
+IFACES=""
+
+NGINX_CHANGED=0
+
+
+config_allowed_hosts_indirectly() {
+    option_cb() { return; }
+
+    list_cb() {
+        [ "$1" = "allow_all_ips_of" ] && {
+            local iface="${2//[^0-9A-Za-z_]}"
+
+            [ "$2" = "${iface}" ] \
+            || echo "Warning: allow_all_ips_of '$2' invalid, using ${iface} ..."
+
+            IFACES="${IFACES} ${iface}"
+
+            for ip in \
+                $(ifstatus $iface | jsonfilter -e '@["ipv4-address"].*.address')
+            do INI="${INI}allowed_host_${ip//[^0-9]/_} = ${ip}\n"; done
+
+            for ip in \
+                $(ifstatus $iface | jsonfilter -e '@["ipv6-address"].*.address')
+            do INI="${INI}allowed_host_${ip//[^0-9A-Fa-f]/_} = [${ip}]\n"; done
+
+            for ip in $(ifstatus $iface | jsonfilter \
+                -e '@["ipv6-prefix-assignment"].*["local-address"].address')
+            do INI="${INI}allowed_host_${ip//[^0-9A-Fa-f]/_} = [${ip}]\n"; done
+
+            return
+        }
+
+        [ "$1" = "allowed_host" ] && {
+            local domain="${2//[^0-9A-Za-z.-]}"
+
+            [ "$2" = "${domain}" ] \
+            || echo "Warning: allowed_host '$2' invalid, using ${domain} ..."
+
+            INI="${INI}allowed_host_${domain//[.-]/_} = ${domain}\n"
+
+            return
+        }
+    }
+}
+
+
+config_cb() {
+    local type="$1"
+    local name="$2"
+    if [ -n "$type" ] || [ -n "$name" ]
+    then
+        INI="${INI}\n[${name}]\n";
+
+        if [ "$type" = "indirect" ] && [ "$name" = "allowed_hosts" ]
+        then config_allowed_hosts_indirectly
+        else option_cb() { INI="${INI}$1 = $2\n"; }
+        fi
+    else
+        reset_cb
+
+        echo "$INI" | sed 's/\\n/\n/g' > /var/etc/etesync-server.ini
+
+        [ -x /etc/init.d/nginx ] || exit 1
+
+        if /etc/init.d/nginx running
+        then
+            [ "${NGINX_CHANGED}" -ne 0 ] \
+            && echo "Reloading Nginx with changed configuration ..." \
+            && /etc/init.d/nginx reload
+        else
+            echo "Starting Nginx ..."
+            /etc/init.d/nginx start
+        fi
+    fi
+}
+
+
+start_service() {
+    rm -f /var/etc/etesync-server.ini
+    config_load etesync_server
+
+    [ -e /usr/share/etesync-server/etesync-server.ini ] \
+    || ln -s /var/etc/etesync-server.ini /usr/share/etesync-server/
+
+    cd /etc/uwsgi/vassals/ && ln -s etesync-server.uwsgi etesync-server.ini
```

Comment:
I think signals should work (also with the emperor) I will look into that :-)

Reply:
In the on-demand mode of the emperor stopping is not really working with signals, I am using the ln/touch functionality again, but now on tmpfs; so, the flash will not wear.

Reply:
There must be a way to make uwsgi react to signals... Locally I use the `touch etesync-server.ini` method to restart it though. :)

Reply:
There is a way you can send SIGINT and co to it (uwsgi --reload and uwsgi --stop are working, too). But, if it is a on-demand vassal of the emperor, it will be reloaded on the next request.
The `touch/ln/rm` mechanism is working also in this setup and it should be no problem anymore as it is now on `/var=tmpfs` (beside it is a bit more complicated in the sense that there is one more symlink).